### PR TITLE
nixUnstable: fix cross by disabling documentation

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

The documentation for Nix HEAD now requires running the generated Nix
binary, so we can't cross-build it. :(

https://github.com/NixOS/nix/issues/4622

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](./CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -36,16 +38,20 @@ common =
 
       VERSION_SUFFIX = suffix;
 
-      outputs = [ "out" "dev" "man" "doc" ];
+      outputs =
+        [ "out" "dev" ]
+        ++ lib.optionals enableDocumentation [ "man" "doc" ];
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/nix/default.nix:43]:
```suggestion
      outputs = [ "out" "dev" ]
        ++ lib.optionals enableDocumentation [ "man" "doc" ];
```

**Comment by SuperSandro2000** [comment]
@alyssais 
```
nix-build -A pkgsCross.aarch64-multiplatform.nixUnstableerror: Package ‘nix-2.4pre20210601_5985b8b’ in /home/hotpi/.cache/nixpkgs-review/pr-125772/nixpkgs/pkgs/tools/package-management/nix/default.nix:160 has invalid meta.outputsT
oInstall, refusing to evaluate.
       The package nix-2.4pre20210601_5985b8b has set meta.outputsToInstall to: out, man

       however nix-2.4pre20210601_5985b8b only has the outputs: out, dev

       and is missing the following ouputs:

         - man
(use '--show-trace' to show detailed location information)
```

File: pkgs/tools/package-management/nix/default.nix (hunk 1 above)
Comment (alyssais) [reply pkgs/tools/package-management/nix/default.nix:43]: This would be inconsistent with the style of the rest of this file.

**Comment by alyssais** [comment]
Yikes, kind of unfortunate that there's no way to catch that using stable Nix...

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 125772` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>hydra-unstable</li>
<li>nix-direnv</li>
<li>nix-update</li>
<li>nixUnstable</li>
<li>nixpkgs-review</li>
</ul>
</details>

**Comment by alyssais** [comment]
It is possible to catch it, turns out.  It's a Nixpkgs feature, not a Nix one.  I just needed to turn on checkMeta. :)

**Pull request merged**